src/main/java/ui/services/CycleTimeService.java

```diff
@@ -116,12 +150,42 @@ protected Task<Object> createTask() {
         return new Task<>() {
             @Override
             protected Object call() throws Exception {
-                if (sprint == null) {
+                if (startDate == null || endDate == null) {
                     return null;
                 }
+                rawTasks.clear();
```

Comment:
Any reason for clearing separately, rather than using `setAll` which clears and adds in the same call? As much as possible, we should discourage `addAll` in certain contexts.

Reply:
Dead as this is removed with implementation of local variables